backend/stakepoold/stakepool/stakepool.go

```diff
@@ -356,15 +356,82 @@ func (spd *Stakepoold) AddMissingTicket(ticketHash []byte) error {
 	return nil
 }
 
-// ListScripts performs the rpc command listscripts on dcrwallet.
+// ListScripts performs the rpc command validateaddress on all known addresses
+// and returns a list of redeem scripts.
 func (spd *Stakepoold) ListScripts() ([][]byte, error) {
-	scripts, err := spd.WalletConnection.RPCClient().ListScripts()
-	if err != nil {
-		log.Errorf("ListScripts: ListScripts rpc failed: %v", err)
-		return nil, err
+	const op = "ListScripts"
+	spd.Lock()
+	addrs := make([]dcrutil.Address, 0, len(spd.UserVotingConfig))
+	// UserVotingConfig should have multi sig addresses for all users.
+	for addr := range spd.UserVotingConfig {
+		decodedAddress, err := dcrutil.DecodeAddress(addr, spd.Params)
+		if err != nil {
+			spd.Unlock()
+			err = fmt.Errorf("%s: failed to decode address %s: %v", op, addr, err)
+			log.Error(err)
+			return nil, err
+		}
+		addrs = append(addrs, decodedAddress)
 	}
-
-	return scripts, nil
+	spd.Unlock()
+	type result struct {
+		addr, hex string
+		err       error
+	}
+	results := make(chan *result)
+	var wg sync.WaitGroup
+	wg.Add(len(addrs))
+	// listScripts validates a slice of addresses using RPC calls.
+	listScripts := func(addrs []dcrutil.Address) {
+		for _, addr := range addrs {
+			defer wg.Done()
+			validated, err := spd.WalletConnection.RPCClient().ValidateAddress(addr)
+			if err != nil {
+				err = fmt.Errorf("failed to validate address %s", addr.String())
+				log.Error("%s: %v", op, err)
+				results <- &result{err: err}
+				continue
+			}
+			results <- &result{addr: addr.String(), hex: validated.Hex}
```

Comment:
Why is this using `validated.Hex` rather than `validated.Script`?

Reply:
iirc the type of script is there, not the script, for example, "multisig"

Reply:
https://github.com/decred/dcrwallet/blob/270804fbd664b5cb124400c3ea10dcdb21bb51de/internal/rpc/jsonrpc/methods.go#L3620